src/System.Security.Cryptography.X509Certificates/tests/CtorTests.cs

```diff
@@ -67,26 +96,130 @@ public static void TestByteArrayConstructor_PEM()
+            };
+
+            using (X509Certificate2 c = new X509Certificate2(TestData.MsCertificatePemBytes))
+            {
+                assert(c);
+#if netstandard17
+                using (X509Certificate2 c2 = new X509Certificate2(c))
+                {
+                    assert(c2);
+                }
+#endif
+            }
+        }
+
+#if netstandard17
+        [Fact]
+        public static void TestCopyConstructor_NoPal()
+        {
+            using (var c1 = new X509Certificate2())
+            using (var c2 = new X509Certificate2(c1))
+            {
+                VerifyDefaultConstructor(c1);
+                VerifyDefaultConstructor(c2);
+            }
+        }
+
+        [Fact]
+        public static void TestCopyConstructor_Pal()
+        {
+            using (var c1 = new X509Certificate2(TestData.PfxData, TestData.PfxDataPassword))
+            using (var c2 = new X509Certificate2(c1))
+            {
+                Assert.Equal(c1.GetCertHash(), c2.GetCertHash());
+                Assert.Equal(c1.GetKeyAlgorithm(), c2.GetKeyAlgorithm());
+                Assert.Equal(c1.GetKeyAlgorithmParameters(), c2.GetKeyAlgorithmParameters());
+                Assert.Equal(c1.GetKeyAlgorithmParametersString(), c2.GetKeyAlgorithmParametersString());
+                Assert.Equal(c1.GetPublicKey(), c2.GetPublicKey());
+                Assert.Equal(c1.GetSerialNumber(), c2.GetSerialNumber());
+                Assert.Equal(c1.Issuer, c2.Issuer);
+                Assert.Equal(c1.Subject, c2.Subject);
+                Assert.Equal(c1.RawData, c2.RawData);
+                Assert.Equal(c1.Thumbprint, c2.Thumbprint);
+                Assert.Equal(c1.SignatureAlgorithm.Value, c2.SignatureAlgorithm.Value);
+                Assert.Equal(c1.HasPrivateKey, c2.HasPrivateKey);
+                Assert.Equal(c1.Version, c2.Version);
+                Assert.Equal(c1.Archived, c2.Archived);
+                Assert.Equal(c1.SubjectName.Name, c2.SubjectName.Name);
+                Assert.Equal(c1.IssuerName.Name, c2.IssuerName.Name);
+                Assert.Equal(c1.GetCertHashString(), c2.GetCertHashString());
+                Assert.Equal(c1.GetEffectiveDateString(), c2.GetEffectiveDateString());
+                Assert.Equal(c1.GetExpirationDateString(), c2.GetExpirationDateString());
+                Assert.Equal(c1.GetPublicKeyString(), c2.GetPublicKeyString());
+                Assert.Equal(c1.GetRawCertData(), c2.GetRawCertData());
+                Assert.Equal(c1.GetRawCertDataString(), c2.GetRawCertDataString());
+                Assert.Equal(c1.GetSerialNumberString(), c2.GetSerialNumberString());
+#pragma warning disable 0618
+                Assert.Equal(c1.GetIssuerName(), c2.GetIssuerName());
+                Assert.Equal(c1.GetName(), c2.GetName());
+#pragma warning restore 0618
+            }
+        }
+
+        [Fact]
+        public static void TestCopyConstructor_Lifetime()
```

Comment:
>> Can you please add a test for disposing in the reverse order as well?

Done

>> It also doesn't look like you added the test Jeremy suggested (creating two separate objects from the same PFX and confirming their lifetimes are independent)

I read Jeremy's comment as being what I added as TestCopyConstructor_Lifetime() meaning the second cert was created with the first cert otherwise I don't see how I could have broken that. Note that some of the commit history was lost due to rebasing (I added TestCopyConstructor_Lifetime() after Jeremy's comment). However, I did add a new test verbatim that will test completely independent objects.

>>Is it also possible to test that when everything is finalized after duplication that native objects do get cleaned up?

Good suggestion however I don't think I have a good way to test that without reflection and\or using a gc.collect pattern.